src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2756,9 +2772,18 @@ private void recordRelocation(void* location, void* target, ushort fRelocType, u
             // relocDelta is stored as the value
             Relocation.WriteValue(relocType, location, relocDelta);
 
-            if (_relocs.Count == 0)
-                _relocs.EnsureCapacity(_code.Length / 32 + 1);
-            _relocs.Add(new Relocation(relocType, relocOffset, relocTarget));
+            if (locationBlock == BlockType.Code)
+            {
+                if (_codeRelocs.Count == 0)
```

Comment:
Nit: could use a `ref ArrayBuilder<Relocation> sourceBlock;` local variable that either points to `_codeRelocs` or _roDataRelocs` to avoid the duplicated code.

Reply:
Thank you. I fixed but I am not sure I did in right way.